receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -209,7 +208,9 @@ func (p *StatsDParser) Aggregate(line string) error {
 	case CounterType:
 		_, ok := p.counters[parsedMetric.description]
 		if !ok {
-			p.counters[parsedMetric.description] = buildCounterMetric(parsedMetric, p.isMonotonicCounter, timeNowFunc(), p.lastIntervalTime)
+			timeNow := timeNowFunc()
+			p.counters[parsedMetric.description] = buildCounterMetric(parsedMetric, p.isMonotonicCounter, timeNow, p.lastIntervalTime)
+			p.lastIntervalTime = timeNow
```

Comment:
Get the first time of the aggregation, then put it to the parser's lastIntervalTime for the next use.

Reply:
Let me take a look one more time on that and come back later. 

So far, I only see one issue that we have a wrapper for a one-line function

```go
var timeNowFunc = func() time.Time {
	return time.Now()
}
```
I mean like we could just use time.Now() directly :D